### PR TITLE
ported to version 2.3.2 and added self_timeout command

### PR DESCRIPTION
**Pull request by Kronifer**
see title

### CODE REVIEW

File: Cogs/self_timeout.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+class SelfTimeout(BaseCog):
+    def __init__(self, bot):
+        BaseCog.__init__(self, bot)
+        self.bot = bot
+    
+    @commands.command(aliases=["selftimeout", "to"])
+    async def selfto(self, ctx, duration: str = commands.parameter(description="Length of time out. Examples: 2m = 2 minutes, 5h = 5 hours.")):
+        """Times the user out for a specified amount of time which cannot exceed 48 hours. Go do your work!"""
+        BaseCog.check_not_private(self, ctx)
+        if duration[-1].lower() not in ["h", "m"] or len(duration) <= 1:
```
Comment (Lustidrike) [inline Cogs/self_timeout.py]: Shouldn't you be checking for the empty string before indexing?
I.e `if len(duration) <= 1 or duration[-1].lower() not in ["h", "m"]:`

I haven't tested but I assume that an empty string would throw an exception in the current code version.

**Review by Lustidrike — COMMENTED** [review]
Hi, this looks great! I had one small comment, everything else looks fine. Appreciate the time you took to port to the latest discord.py version.

**Comment by Kronifer** [comment]
should all be resolved

**Review by Lustidrike — APPROVED** [review]
Looks great, thanks!

**Pull request merged**